receiver/statsdreceiver/testutil/temporary_socket.go

```diff
@@ -0,0 +1,13 @@
+package testutil
+
+import (
+	"crypto/rand"
+	"fmt"
+	"testing"
+)
+
+func CreateTemporarySocket(t testing.TB, _ string) string {
```

Comment:
why do we have the second argument if we ignore it?

Reply:
isn't easier to have a fixed name if tempDir() is always changing?

Reply:
or you can leverage createTemp https://pkg.go.dev/os#CreateTemp

Reply:
Because the signature has to be compatible with https://github.com/kang-makes/opentelemetry-collector-contrib/blob/receiver/statsdreceiver/add-uds-support-part-2/internal/common/testutil/testutil.go#L35

I also used `t.TempDir()` because the testing suite cleans all files after finishing tests so we can keep the house clean :)